Add identity matrix generation functionality to SimpleArray (#588)

tkoyama010 opened the pull request:
## Summary
This PR implements identity matrix generation functionality for SimpleArray, addressing one of the matrix operations listed in issue #588.

## Changes
- ✅ **C++ Implementation**: Added `SimpleArray::eye(size_t n)` static method
- ✅ **Python Bindings**: Added `.eye(n)` method with proper documentation
- ✅ **Integration**: Updated existing GEMM tests to use new `eye` method
- ✅ **Multi-type Support**: Works with Float32 and Float64 SimpleArray types

## Usage
```python
import modmesh as mm

# Generate 3x3 identity matrix
identity = mm.SimpleArrayFloat64.eye(3)
print(identity.ndarray)
# [[1. 0. 0.]
#  [0. 1. 0.]
#  [0. 0. 1.]]

# Use with matrix multiplication
result = matrix.matmul(mm.SimpleArrayFloat64.eye(3))  # result equals matrix
```

Closes part of #588

tkoyama010 committed:
Add identity matrix generation functionality to SimpleArray

- Implement SimpleArray::eye(n) static method for generating n×n identity matrices
- Add Python bindings with proper documentation
- Create comprehensive test suite with 20 unit test cases
- Update existing GEMM tests to use new eye method
- Support both Float32 and Float64 types
- Follow NumPy interface conventions
- All tests pass, no breaking changes

Addresses GitHub issue #588: Generate identity matrix

tkoyama010 committed:
Delete tests/test_identity_matrix.py

tkoyama010 commented on tests/test_gemm.py:
```diff
@@ -75,10 +75,9 @@ def test_identity_matrix(self):
+        
         a = self.SimpleArray(array=a_data)
-        identity = self.SimpleArray(array=identity_data)
+        identity = self.SimpleArray.eye(3)  # Use our new eye method
```
```suggestion
        identity = self.SimpleArray.eye(3)
```

tkoyama010 committed:
Update tests/test_gemm.py

yungyuc commented on tests/test_gemm.py:
Please also create a distinct test function for `eye` in addition to exercise it inside the test function for matmul.  Both are good and the distinct test function will help clarify bug root causes in the future.

tkoyama010 replied:
resolve in 089586d8946163c8ddbcc0d0c36e5acff1e056a6

yungyuc reviewed (changes requested):
Thanks for the contribution.  Points to address:

- [ ] Distinct test function for `eye()`.
- [ ] Make sure CI pass.

tkoyama010 committed:
Fix linting issues: remove trailing whitespace

- Remove trailing whitespace from C++ comment and code lines
- Remove trailing whitespace from Python test file
- Fixes clang-format and flake8 linting errors

tkoyama010 committed:
Add dedicated test function for eye method

- Add test_eye_method() as requested in review comment
- Uses subTest for parameterized testing with sizes [1,2,3,4,5,10]
- Uses itertools.product for efficient nested loop iteration
- Tests both array comparison and individual element verification
- Provides better test isolation for debugging future issues

Addresses review comment: Create distinct test function for eye method
in addition to exercising it in matmul test function.

tkoyama010 committed:
Fix flake8 linting errors in test_gemm.py

- Remove trailing whitespace from blank lines
- Fix line length violations by breaking long lines appropriately
- Fix indentation for continuation lines
- Maintain functionality while ensuring PEP 8 compliance

yungyuc reviewed (approved):
Thank you.  LGTM.
